File: code/platform-server/platform-user/src/main/java/com/cloudhub/platform/user/domain/entity/Dept.java
package com.cloudhub.platform.user.domain.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.cloudhub.platform.common.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_dept")
public class Dept extends BaseEntity {
    
    private Long orgId;            // 所属组织ID
    private Long parentId;         // 父部门ID
    private String name;           // 部门名称
    private String code;           // 部门编码
    private String manager;        // 部门负责人
    private String phone;          // 联系电话
    private Integer sort;          // 排序
    private Integer status;        // 状态
    
    @TableField(exist = false)
    private List<Dept> children = new ArrayList<>();
}

File: code/platform-admin/src/api/dept.ts
import request from './request'

export interface Dept {
  id?: string
  orgId?: number
  parentId?: string
  name?: string
  code?: string
  manager?: string
  phone?: string
  sort?: number
  status?: number
  children?: Dept[]
}

export function getDeptTree(orgId?: number) {
  return request({
    url: '/dept/tree',
    method: 'get',
    params: orgId ? { orgId } : {}
  })
}

export function getDeptList(orgId: number | string) {
  return request({
    url: `/dept/org/${orgId}`,
    method: 'get'
  })
}

export function getDeptById(id: string) {
  return request({
    url: `/dept/${id}`,
    method: 'get'
  })
}

export function createDept(data: Dept) {
  return request({
    url: '/dept',
    method: 'post',
    data
  })
}

export function updateDept(id: string, data: Dept) {
  return request({
    url: `/dept/${id}`,
    method: 'put',
    data
  })
}

export function deleteDept(id: string) {
  return request({
    url: `/dept/${id}`,
    method: 'delete'
  })
}